### ISSUE TITLE
SSSP on `soc-orkut` dataset

### ISSUE THREAD

**Issue by bkj** [gunrock/gunrock#414]
Running:
```
bin/test_sssp_9.1_x86_64 --graph-type market --graph-file /home/bjohnson/projects/davis/gunrock/dataset/large/soc-orkut/soc-orkut.mtx --src 10 --num-runs 1
```
throws
```
Loading Matrix-market coordinate-formatted graph ...
Reading from /home/bjohnson/projects/davis/gunrock/dataset/large/soc-orkut/soc-orkut.mtx:
  Parsing MARKET COO format edge-value-seed = 1539711772
 (2997166 nodes, 106349209 directed edges)... 
Done parsing (32 s).
  Converting 2997166 vertices, 106349209 directed edges ( ordered tuples) to CSR format...
Done converting (0s).
Computing reference value ...
__________________________
--------------------------
Run 0 elapsed: 0.001907 ms, src = 10
==============================================
 mark-pred=0 advance-mode=LB
Using advance mode LB
Using filter mode CULL
__________________________
[../../gunrock/util/array_utils.cuh, 888 @ gpu 0] edges_queued cudaMemcpy D2H failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/util/array_utils.cuh, 888 @ gpu 0] nodes_queued cudaMemcpy D2H failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/enactor_loop.cuh, 686 @ gpu 0] FullQueue_Core failed. (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/enactor_base.cuh, 568 @ gpu 0] error encountered (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/sssp/sssp_enactor.cuh, 385 @ gpu 0] error encountered (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/sssp/sssp_app.cu, 117 @ gpu 0] error encountered (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/util/array_utils.cuh, 536 @ gpu 0] distances cudaFree failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/sssp/sssp_problem.cuh, 113 @ gpu 0] error encountered (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/util/array_utils.cuh, 536 @ gpu 0] data_slices[0] cudaFree failed (CUDA error 77: an illegal memory access was encountered)
[../../gunrock/app/sssp/sssp_problem.cuh, 239 @ gpu 0] error encountered (CUDA error 77: an illegal memory access was encountered)
[test_sssp.cu, 101 @ gpu 0] error encountered (CUDA error 77: an illegal memory access was encountered)
```
The app runs on other datasets.  Any ideas what might be going on here?  Are we able to run other apps on the `soc-orkut` dataset?

~ Ben

**Comment by sgpyc** [comment]
I can reproduce the bug on top of dev-refactor branch.
It should be easy to track down, but may need to push it after the beginning of Nov.

PageRank works on soc-orkut

**Comment by sgpyc** [comment]
This issue is caused by the COO->CSR or COO->CSC converter, when the first vertices don't have any neighbor. Solved by commit 1f5b83c
```
./bin/test_sssp_10.0_x86_64 market /data/gunrock_dataset/large/soc-orkut/soc-orkut.mtx --device=1 --src=10
Loading Matrix-market coordinate-formatted graph ...
  Reading meta data from /data/gunrock_dataset/large/soc-orkut/soc-orkut.mtx.meta
  Reading edge lists from /data/gunrock_dataset/large/soc-orkut/soc-orkut.mtx.coo_edge_pairs
  Assigning 1 to all 106349209 edges
  Substracting 1 from node Ids...
  graph loaded as COO in 5.389060s.
Converting 2997166 vertices, 106349209 directed edges ( ordered tuples) to CSR format...Done (0s).
Degree Histogram (2997166 vertices, 106349209 edges):
    Degree 0: 1005 (0.033532 %)
    Degree 2^0: 92480 (3.085582 %)
    Degree 2^1: 158955 (5.303510 %)
    Degree 2^2: 316144 (10.548098 %)
    Degree 2^3: 561754 (18.742839 %)
    Degree 2^4: 766179 (25.563449 %)
    Degree 2^5: 672335 (22.432358 %)
    Degree 2^6: 320814 (10.703912 %)
    Degree 2^7: 85094 (2.839149 %)
    Degree 2^8: 20065 (0.669466 %)
    Degree 2^9: 2236 (0.074604 %)
    Degree 2^10: 100 (0.003336 %)
    Degree 2^11: 5 (0.000167 %)

Computing reference value ...
__________________________
--------------------------
Run 0 elapsed: 0.004053 ms, src = 10
==============================================
 mark-pred=0 advance-mode=LB
Using advance mode LB
Using filter mode CULL
__________________________
--------------------------
Run 0 elapsed: 0.406981 ms, src = 10, #iterations = 3
Distance Validity: PASS
First 40 distances of the GPU result:
[0:1 1:4294967295 2:2 3:2 4:4294967295 5:4294967295 6:1 7:1 8:1 9:1 10:0 11:4294967295 12:4294967295 13:4294967295 14:4294967295 15:4294967295 16:4294967295 17:4294967295 18:4294967295 19:4294967295 20:4294967295 21:4294967295 22:4294967295 23:4294967295 24:4294967295 25:4294967295 26:4294967295 27:4294967295 28:4294967295 29:4294967295 30:4294967295 31:4294967295 32:4294967295 33:4294967295 34:4294967295 35:4294967295 36:4294967295 37:4294967295 38:4294967295 39:4294967295 ]
First 40 distances of the reference CPU result.
[0:1 1:4294967295 2:2 3:2 4:4294967295 5:4294967295 6:1 7:1 8:1 9:1 10:0 11:4294967295 12:4294967295 13:4294967295 14:4294967295 15:4294967295 16:4294967295 17:4294967295 18:4294967295 19:4294967295 20:4294967295 21:4294967295 22:4294967295 23:4294967295 24:4294967295 25:4294967295 26:4294967295 27:4294967295 28:4294967295 29:4294967295 30:4294967295 31:4294967295 32:4294967295 33:4294967295 34:4294967295 35:4294967295 36:4294967295 37:4294967295 38:4294967295 39:4294967295 ]

[SSSP] finished.
 avg. elapsed: 0.406981 ms
 iterations: 3
 min. elapsed: 0.406981 ms
 max. elapsed: 0.406981 ms
 rate: 0.039314 MiEdges/s
 src: 10
 nodes_visited: 8
 edges_visited: 16
 load time: 5742.12 ms
 preprocess time: 1115.470000 ms
 postprocess time: 14.773130 ms
 total time: 1130.831003 ms
```

**Comment by neoblizz** [comment]
Assuming this is resolved!
